src/rust/engine/options/src/parse.rs

```diff
@@ -55,23 +55,51 @@ peg::parser! {
             = "\"" s:double_quoted_character()* "\"" { s.into_iter().collect() }
 
         rule double_quoted_character() -> char
-            = quoted_character("\"")
-            / escaped_character()
+            = escaped_character() / quoted_character("\"")
 
         rule single_quoted_string() -> String
             = "'" s:single_quoted_character()* "'" { s.into_iter().collect() }
 
         rule single_quoted_character() -> char
-            = quoted_character("'")
-            / escaped_character()
+            = escaped_character() / quoted_character("'")
 
         // NB: ##method(X) is an undocumented peg feature expression that calls input.method(pos, X)
         // (see https://github.com/kevinmehall/rust-peg/issues/283).
         rule quoted_character(quote_char: &'static str) -> char
-            = !(##parse_string_literal(quote_char) / "\\") c:$([_]) { c.chars().next().unwrap() }
+            = !(##parse_string_literal(quote_char)) c:$([_]) { c.chars().next().unwrap() }
+
+        // Python string literal escape sequences.
+        // See https://docs.python.org/3/reference/lexical_analysis.html#escape-sequences.
+        // Note that only backslash, single-quote, linefeed, carriage return and horizontal tab
+        // are also Rust character escape sequences.
+
+        rule escaped_ba() -> char = "\\\\" { '\\' }
+        rule escaped_sq() -> char = "\\'" { '\'' }
+        rule escaped_dq() -> char = "\\\"" { '"' }
+        rule escaped_be() -> char = "\\a" { '\x07' }
+        rule escaped_bs() -> char = "\\b" { '\x08' }
+        rule escaped_ff() -> char = "\\f" { '\x0c' }
+        rule escaped_lf() -> char = "\\n" { '\n' }
+        rule escaped_cr() -> char = "\\r" { '\r' }
+        rule escaped_ht() -> char = "\\t" { '\t' }
+        rule escaped_vt() -> char = "\\v" { '\x0b' }
+
+        // Python octal escapes take 1-3 digits.
+        rule escaped_octal() -> char = "\\" s:$(['0'..='7']*<1,3>) {
+            char::from_u32(usize::from_str_radix(s, 8).unwrap().try_into().unwrap()).unwrap()
```

Comment:
Does this work?

```suggestion
            char::from_u32(u32::from_str_radix(s, 8).unwrap()).unwrap()
```

Reply:
It does! much nicer. For some reason I missed that in the u32 docs.